.golangci-kal.yml

```diff
@@ -0,0 +1,55 @@
+run:
+  timeout: 10m
+  go: "1.23"
+  allow-parallel-runners: true
+
+linters:
+  disable-all: true
+  enable:
+    - kal # linter for Kube API conventions
+   
+linters-settings:
+  custom:
+    kal:
+      type: "module"
+      description: KAL is the Kube-API-Linter and lints Kube like APIs based on API conventions and best practices.
+      settings:
+        linters:
+          enable:
+          - "conditions" # Ensure conditions have the correct json tags and markers.
+          - "commentstart" # Ensure comments start with the serialized version of the field name.
+          - "integers" # Ensure only int32 and int64 are used for integers.
+          - "jsontags" # Ensure every field has a json tag.
+          - "maxlength" # Ensure all strings and arrays have maximum lengths/maximum items.
+          - "nobools" # Bools do not evolve over time, should use enums instead.
+          - "nophase" # Phase fields are discouraged by the Kube API conventions, use conditions instead.
+          - "optionalorrequired" # Every field should be marked as `+optional` or `+required`.
+          - "requiredfields" # Required fields should not be pointers, and should not have `omitempty`.
+          - "statussubresource" # All root objects that have a `status` field should have a status subresource.
+          disable:
+          - "*" # We will manually enable new linters after understanding the impact. Disable all by default.
+        lintersConfig:
+          conditions:
+            isFirstField: Warn # Require conditions to be the first field in the status struct.
+            useProtobuf: Ignore # We don't use protobuf, so missing protobuf tags are ignored.
+        # jsonTags:
+        #   jsonTagRegex: "^[a-z][a-z0-9]*(?:[A-Z][a-z0-9]*)*$" # The default regex is appropriate for our use case.
+        # optionalOrRequired:
+        #   preferredOptionalMarker: optional | kubebuilder:validation:Optional # The preferred optional marker to use, fixes will suggest to use this marker. Defaults to `optional`.
```

Comment:
Definitely in favor of using required / optional over the kubebuilder markers

Reply:
In general, I think using the "upstream" markers vs kubebuilder markers should be our stance, so, in agreement. I know that the `default` marker is now supported by controller-gen too so avoiding the kubebuilder default marker variation would be good. There's a todo to implement a marker that points this out, but it's not implemented yet